Editor/Drawers/ReferenceDrawer.cs

```diff
@@ -182,18 +221,26 @@ private void HandleDragPerform()
             switch (dragAndDropMode)
             {
                 case DragAndDropMode.Raw:
-                    RawReferenceProperty.managedReferenceValue =
-                        Activator.CreateInstance(((MonoScript)DragAndDrop.objectReferences[0]).GetClass());
-                    ReferenceModeProperty.enumValueIndex = (int)ReferenceMode.Raw;
+                    ModeValue = ReferenceMode.Raw;
+                    PropertyValue = Activator.CreateInstance(((MonoScript)DragAndDrop.objectReferences[0]).GetClass());
                     break;
                 case DragAndDropMode.Unity:
-                    UnityReferenceProperty.objectReferenceValue = DragAndDrop.objectReferences[0];
-                    ReferenceModeProperty.enumValueIndex = (int)ReferenceMode.Unity;
+                    ModeValue = ReferenceMode.Unity;
+                    PropertyValue = GetUnityObject(DragAndDrop.objectReferences[0]);
```

Comment:
This PR also fixes another bug with Drag and Dropping a GameObject that contained a Component with the right interface on it. 
Previously, it would take DragAndDrop.objectReferences[0], which would be a GameObject, and set it to the UnityReferenceProperty.objectReferenceValue. This would work in the inspector, but at runtime, it would throw a InvalidCastException as it would try to cast the GameObject to the interface.
The solution was simply to get the right component instead of the GameObject.